production.py

```diff
@@ -380,14 +383,17 @@ def calc_group(self, type_, kind):
                                 Decimal(10) ** -price_digits[1])
                 move_cost.total += move.total
 
+        to_delete_ids = [t.id for t in to_delete]
         MoveCost.delete(to_delete)
-        return res.values()
+        return res.values(), to_delete_ids
 
     def calc_group_operation(self, kind):
         pool = Pool()
         OperationCost = pool.get('production.cost.analysis.operation')
+
         to_delete = [x for x in self.operation_costs if not x.operation
             and x.kind == kind]
+
         res = {}
         for op in self.operation_costs:
             if not op.operation or op.kind != kind:
```

Comment:
The actual problem is that the condition used to compute to_delete is different from the one on the if at the beginning of the loop (one uses 'and' the other uses 'or').

However, instead of changing and/or I think that we avoid duplicate code and potential mistakes if we use:

```
for op in self.operation_costs:
    if op not in to_delete:
        continue
... 
```

Reply:
By the way, the same problem exists at calc_group() method.

Given that those conditions are so different (due to add/or), I think we should review that the condition in to_delete is the correct one.

Reply:
> for op in self.operation_costs:
>     if op not in to_delete:
>         continue
> ...

I don't know the condition _to_delete_ and the condition to _continue_ because use _and_ and _or_.

See calc_group() method not delete cancelled moves, but yes continue cancelled moves.